nncf/torch/external_hook.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+
+from nncf.torch.dynamic_graph.context import TracingContext
+
+EXTERNAL_OP_STORAGE_NAME = "external_op"
```

Comment:
In view of the more general "op" now being introduced, does it make sense still to keep the "external_quantizer" key? If not, prepare the transition of the "external_quantizer" state dict key to "external_op", with backward compatibility and associated deprecation notices.

Reply:
Also, the external op storage name belongs to NNCFNetwork, not to the hook itself.

Reply:
That's nice idea, I think the best way to progress with that right now is to add it to refactoring list. That will be relevant when QAT initialization will be migrating to PTQ